addon_service/credentials/encryption.py

```diff
@@ -0,0 +1,142 @@
+"""for encrypting and decrypting data (using secrets from settings)
+
+considering recommendations from:
+- https://cryptography.io/en/latest/fernet/
+- https://docs.python.org/3.12/library/hashlib.html#hashlib.scrypt
+- https://cryptography.io/en/latest/hazmat/primitives/key-derivation-functions/#scrypt
+- https://datatracker.ietf.org/doc/html/rfc7914
+- https://nvlpubs.nist.gov/nistpubs/Legacy/SP/nistspecialpublication800-132.pdf
+"""
+
+import base64
+import dataclasses
+import functools
+import hashlib
+import json
+import math
+import os
+
+from cryptography import fernet
+from django.conf import settings
+
+
+__all__ = (
+    "pls_decrypt_bytes",
+    "pls_decrypt_json",
+    "pls_encrypt_bytes",
+    "pls_encrypt_json",
+    "salt_factory",
+)
+
+
+# 32-byte key expected by cryptography.fernet.Fernet
+_KEY_BYTE_COUNT = 32
+
+# recommended len(salt) >= 16 bytes
+_SALT_BYTE_COUNT = settings.GRAVYVALET_SALT_BYTE_COUNT or 17
```

Comment:
Should we also make note in the settings that `settings.GRAVYVALET_SALT_BYTE_COUNT` is recommended to be >= 16?

Reply:
we should enforce that, probably... i've added `len(self.salt) >= 16` to what's validated in `KeyParameters.__post_init__` -- if/when we document the other parameters outside this file, should include that constraint